src/sales.ts

```diff
@@ -79,14 +80,14 @@ export const getSales = (api: ApiInstance) => ({
   return api.get('/vendas/getFilterBy', { query });
 };
 
-const mapSale = (sale: Sale): object => ({
-  ...sale,
+const mapSale = evolve({
   produto: {
-    ...sale.produto,
-    tags: joinWithSemicolon(sale.produto.tags),
+    tags: joinWithSemicolon,
   },
-  tags: joinWithSemicolon(sale.tags || []),
+  tags: joinWithSemicolon,
 });
 
 export const createSale = (api: ApiInstance) => (sale: Sale): got.GotPromise<object> =>
```

Comment:
If API return and our object was similar, maybe will b e a better approach compose type and use it at the place os Object. Something like `SaleCore` `SaleApiObject` `Sale`, and make something like SaleCore + App Properties = Sale, SaleCore + Api props = SaleApiObject.

```typescript
const SaleApiReturn = GotPromise<SaleApiObject>
```

```typescript
export const createSale = (api: ApiInstance) => (sale: Sale): SaleApiReturn
```